File: assignments/solutions/01-flow-of-program/leapYear.js
const readline = require('readline');
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

const isLeapYear = (year) => {
    return year % 4 === 0;
};

rl.question('Enter a year to check whether it is Leap Year or not: ', year => {
    isNaN(Number(year)) ? console.log('Please enter a valid Year') :
        console.log(`${year} is ${isLeapYear(year) ? '' : 'not '}a Leap Year`);
    rl.close();
});

/**
 * Output:
 * 
 * Q: Enter a year to check whether it is Leap Year or not: 2022
 * A: 2022 is not a Leap Year
 * 
 * Q: Enter a year to check whether it is Leap Year or not: 2020
 * A: 2020 is a Leap Year
 */

File: assignments/solutions/03-conditionals-loops/leapYear.java
// Leap Year Checker

import java.util.Scanner;

public class leapYear {
    public static void main (String args[]) {
        Scanner scan = new Scanner(System.in);
        System.out.println("Leap Year Checker\n");
        System.out.print("Enter Year: ");
        int year = scan.nextInt();
        boolean isLeapYear = year % 4 == 0;
        System.out.println(year + " is " + (isLeapYear ? "Leap Year" : "not Leap Year"));
    }
}

/*
Output:

Leap Year Checker

Enter Year: 2020
2020 is Leap Year

Enter Year: 2022
2022 is not Leap Year
*/